hawc/apps/epi/serializers.py

```diff
@@ -90,41 +114,70 @@ class Meta:
         fields = "__all__"
 
 
-class SimpleExposureSerializer(serializers.ModelSerializer):
+class ComparisonSetLinkSerializer(serializers.ModelSerializer):
     url = serializers.CharField(source="get_absolute_url", read_only=True)
-    metric_units = DoseUnitsSerializer()
-    central_tendencies = CentralTendencySerializer(many=True)
 
     class Meta:
-        model = models.Exposure
-        fields = "__all__"
+        model = models.ComparisonSet
+        fields = ("id", "name", "url")
 
 
-class ComparisonSetLinkSerializer(serializers.ModelSerializer):
-    url = serializers.CharField(source="get_absolute_url", read_only=True)
+class CriteriaSerializer(GetOrCreateMixin, serializers.ModelSerializer):
+    form_integration_class = forms.CriteriaForm
 
     class Meta:
-        model = models.ComparisonSet
-        fields = ("id", "name", "url")
+        model = models.Criteria
+        fields = "__all__"
+
+
+class SimpleStudyPopulationCriteriaSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = models.StudyPopulationCriteria
+        fields = "__all__"
 
 
-class StudyPopulationSerializer(serializers.ModelSerializer):
+class StudyPopulationSerializer(IdLookupMixin, serializers.ModelSerializer):
+    form_integration_class = forms.StudyPopulationForm
+
     study = StudySerializer()
-    criteria = StudyPopulationCriteriaSerializer(source="spcriteria", many=True)
-    outcomes = OutcomeLinkSerializer(many=True)
-    exposures = ExposureLinkSerializer(many=True)
-    can_create_sets = serializers.BooleanField(read_only=True)
-    comparison_sets = ComparisonSetLinkSerializer(many=True)
+    criteria = StudyPopulationCriteriaSerializer(source="spcriteria", many=True, read_only=True)
     countries = StudyPopulationCountrySerializer(many=True)
+    design = FlexibleChoiceField(choices=models.StudyPopulation.DESIGN_CHOICES)
+    outcomes = OutcomeLinkSerializer(many=True, read_only=True)
+    exposures = ExposureLinkSerializer(many=True, read_only=True)
+    can_create_sets = serializers.BooleanField(read_only=True)
+    comparison_sets = ComparisonSetLinkSerializer(many=True, read_only=True)
     url = serializers.CharField(source="get_absolute_url", read_only=True)
-    design = serializers.CharField(source="get_design_display", read_only=True)
 
     class Meta:
         model = models.StudyPopulation
         fields = "__all__"
 
+    def update(self, instance, validated_data):
+        if "countries" in validated_data:
+            instance.countries.clear()
```

Comment:
instead of clear/add, use set

https://docs.djangoproject.com/en/3.1/ref/models/relations/#django.db.models.fields.related.RelatedManager.set